transforms/universal/pdf2md/python/src/pdf2md_transform.py

```diff
@@ -0,0 +1,346 @@
+        start_time = time.time()
+        buf = io.BytesIO(content_bytes)
+        input_docs = DocumentStream(filename=doc_filename, stream=buf)
+        input = DocumentConversionInput.from_streams([input_docs])
+
+        converted_docs = self._converter.convert(input)
+        doc: ConvertedDocument = next(converted_docs, None)
+        if doc is None or doc.output is None:
+            raise RuntimeError("Failed in converting.")
+        elapse_time = time.time() - start_time
+
+        content_string = doc.render_as_markdown()
+        num_pages = len(doc.pages)
+        num_tables = len(doc.output.tables if doc.output.tables is not None else 0)
+        num_doc_elements = len(doc.output.main_text if doc.output.main_text is not None else 0)
+
+        self._update_metrics(num_pages=num_pages, elapse_time=elapse_time)
+
+        file_data = {
+            "filename": TransformUtils.get_file_basename(doc_filename),
+            "contents": content_string,
+            "num_pages": num_pages,
+            "num_tables": num_tables,
+            "num_doc_elements": num_doc_elements,
+            "document_id": TransformUtils.str_to_hash(content_string),
+            "ext": ext,
+            "hash": TransformUtils.str_to_hash(content_string),
+            "size": len(content_string),
+            "date_acquired": datetime.now().isoformat(),
+            "pdf_convert_time": elapse_time,
+        }
+
+        return file_data
+
+    def transform_binary(
+        self, file_name: str, byte_array: bytes
+    ) -> tuple[list[tuple[bytes, str]], dict[str, Any]]:
+        """
+        If file_name is detected as a PDF file, it generates a pyarrow table with a single row
+        containing the document converted in markdown format.
+        If file_name is detected as a ZIP archive, it generates a pyarrow table with a row
+        for each PDF file detected in the archive.
+        """
+
+        data = []
+        success_doc_id = []
+        failed_doc_id = []
+        skipped_doc_id = []
+        number_of_rows = 0
+
+        try:
+            root_kind = filetype.guess(byte_array)
+
+            # Process single PDF documents
+            if root_kind is not None and root_kind.mime == "application/pdf":
+                logger.debug(
+                    f"Detected root file {file_name=} as PDF."
+                )
+
+                try:
+                    root_ext = root_kind.extension
+                    file_data = self._convert_pdf2md(doc_filename=file_name, ext=root_ext, content_bytes=byte_array)
+
+                    file_data["source_filename"] = TransformUtils.get_file_basename(file_name)
+
+                    data.append(file_data)
+                    number_of_rows += 1
+
+                except Exception as e:
+                    failed_doc_id.append(archive_doc_filename)
+                    logger.warning(
+                        f"Exception {str(e)} processing file {archive_doc_filename}, skipping"
+                    )
+
```

Comment:
So in this case we have a single row table?

Reply:
when this exception triggers, I think the output will be an empty table